ros2-library/src/main/java/us/ihmc/ros2/ROS2NodeBuilder.java

```diff
@@ -0,0 +1,438 @@
+      return this;
+   }
+
+   public ROS2NodeBuilder parseNetworkParametersConfig(boolean parseNetworkParametersConfig)
+   {
+      this.parseNetworkParametersConfig = parseNetworkParametersConfig;
+      return this;
+   }
+
+   public ROS2NodeBuilder specialTransportMode(@Nullable SpecialTransportMode specialTransportMode)
+   {
+      this.specialTransportMode = specialTransportMode;
+      return this;
+   }
+
+   public ROS2Node build(String name)
+   {
+      buildPrintout.add("Building ROS2Node: " + name);
+      return new ROS2Node(name, namespace, buildProfile(), specialTransportMode);
+   }
+
+   public RealtimeROS2Node buildRealtime(String name)
+   {
+      return buildRealtime(name, new PeriodicNonRealtimeThreadSchedulerFactory());
+   }
+
+   public RealtimeROS2Node buildRealtime(String name, PeriodicThreadSchedulerFactory threadFactory)
+   {
+      buildPrintout.add("Building RealtimeROS2Node: " + name);
+      return new RealtimeROS2Node(name, namespace, buildProfile(), specialTransportMode, threadFactory);
+   }
+
+   private ParticipantProfile buildProfile()
+   {
+      ParticipantProfile profile = ParticipantProfile.create();
+
+      // Set up ROS Domain ID
+      {
+         if (domainIDValid(domainId))
+         {
+            buildPrintout.add("Using a programmatically set ROS Domain ID: " + domainId);
+         }
+         else
+         {
+            // Try to find a ROS Domain ID
+            domainId = findDomainID();
+
+            // If a valid domain ID was not found automatically
+            if (!domainIDValid(domainId))
+            {
+               domainId = 100;
+
+               buildPrintout.add("Unable to find any ROS Domain ID");
+               buildPrintout.add(
+                     "You can set a ROS Domain ID via: system property (ros.domain.id), environment variable (ROS_DOMAIN_ID), RTPSDomainID in IHMCNetworkParameters.ini");
+               buildPrintout.add("Using a default ROS Domain ID: " + domainId);
+            }
+         }
+
+         profile.domainId(domainId);
+      }
+
+      // Check namespace and print if it was changed
+      {
+         if (!namespace.equals(DEFAULT_NAMESPACE))
+         {
+            buildPrintout.add("Namespace: " + namespace);
+         }
+      }
+
+      // Set up transports
+      {
+         profile.useBuiltinTransports(false);
+
+         if (useSharedMemory)
+            profile.addSharedMemoryTransport();
+
+         if (addressRestriction != null)
+            buildPrintout.add("Using a programmatically set address restriction");
+         else
+            addressRestriction = findAddressRestriction();
+
+         profile.addUDPv4Transport(addressRestriction);
+
+         boolean runningInCI = (System.getenv("GITHUB_ACTIONS") != null)
+                               || (System.getenv("RUNNING_ON_CONTINUOUS_INTEGRATION_SERVER") != null)
+                               || (System.getProperty("runningOnCIServer") != null);
+         if (runningInCI)
```

Comment:
What about this? Essentially default to intraprocess when in CI, but don't override the special transport mode if it was set by the programmer. This would allow us to easily test shared memory and udp transports too. 
```suggestion
         if (runningInCI && specialTransportMode == null)
```

Reply:
I like this idea, thanks